File: 485-max-consecutive-ones/max-consecutive-ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
   let curCount =0;
   let maxCount =0;
   for(let i=0;i<nums.length;i++){
    if(nums[i]==1){
        curCount++;
        maxCount = Math.max(curCount,maxCount);
    }
    else{
        maxCount = Math.max(curCount,maxCount);
        curCount =0;
    }

   }
   return maxCount;
};

File: 485-max-consecutive-ones/max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int curCount =0;
        int maxCount =0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==0){
                curCount=0;
                maxCount = max(curCount,maxCount);
            }
            else{
                curCount++;
                maxCount = max(curCount,maxCount);
            }
        }
        return maxCount;
    }
};